experiments/process_sandbox/include/process_sandbox/path.h

```diff
@@ -0,0 +1,229 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include "helpers.h"
+
+#include <list>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+namespace sandbox
+{
+  /**
+   * Class encapsulating a path.  A path is route through a tree.  This class
+   * works on an abstract filesystem namespace, it never accesses the
+   * filesystem.
+   */
+  class Path
```

Comment:
Why not use an existing library?

Reply:
Are there any good ones?  I've not seen one that does the things that we actually want.  The `std::filesystem::path` from the Filesystems TS is as much a joke as the rest of that TS, `boost::filesystem::path` expects to operate on the local machine's filesystem and so its canonicalisation also removes symlinks and so doesn't do the right thing for us.

Reply:
Good ones? I doubt it.

LLVM has one (`llvm::sys::fs`), but it's strongly biased to compile unit handling and may not be what you want. I use it in the interop layer for `real_path` and `parent_path`, but that's mostly because the use there is similar to its bias.

The parser has one (`src/parser/path.h`), but it was created for the parser (imported from previous parser), so that, too, has local bias.

Either way, I'd try the LLVM one first, and if that's not good enough, I'd try to make our parser's `path` more generic, than create a third one in our codebase.

Reply:
This one has very specific requirements: It must operate on an abstract filesystem and not ever reference the real filesystem.  Given how simple it is, I'm far more comfortable having a stand-alone implementation here than I am auditing every change to another one to make sure that it does not accidentally provide a means for the sandboxed code to leak data about the filesystem contents.

Reply:
Since this is an experiment, I don't mind much. But once we get this into the code proper, I'd like to fuse this with the parser's own implementation and make it generic. 

Given your requirements, using a third-party one is troublesome. But I'm sure we can have an implementation that we can guarantee doesn't violate our requirements for both parser and interop.

Reply:
It might be possible, but generally I'd want the version for the parser to end up with a capability abstraction, whereas this one is dealing entirely with names.